components/net/http_loader.rs

```diff
@@ -467,3 +465,14 @@ fn send_data<R: Read>(reader: &mut R,
 
     let _ = progress_chan.send(Done(Ok(())));
 }
+
+// FIXME: This incredibly hacky. Make it more robust, and at least test it.
+fn is_cert_verify_error(error: &OpensslError) -> bool {
+    match error {
```

Comment:
Nit: you can use `let` instead of `match` here since there is only one variant.

(This is still hacky, but I feel much better about this than what we could do before `Error::downcast_ref` was added.)